aml-development/ozp-backend
# feat(es): add 508 filter

AMLOS-669
AMLOS-668

**Description**     
Filter search results on 508 flag (669)
Add 508-only-search flag for user profile (668)

**Acceptance Criteria**   
Can filter search results based on 508 flag (669)
Ability to save flag for 508 filter for user (668)
 
**How to test code (669)**    
```
make dev reindex_es use_es run

curl 'http://127.0.0.1:8001/api/listings/essearch/?search=&offset=0&limit=24&is_508_compliant=true'  --user bigbrother:password
```
Results should be 
```
{
  "results": [
    {
      "unique_name": "acoustic_guitar",
      "is_featured": false,
      "tags": [
        {
          "name_string": "acoustic",
          "id": 3,
          "name": "acoustic"
        },
        {
          "name_string": "classical",
          "id": 4,
          "name": "classical"
        },
        {
          "name_string": "concert",
          "id": 5,
          "name": "concert"
        },
        {
          "name_string": "guitar",
          "id": 6,
          "name": "guitar"
        },
        {
          "name_string": "instrument",
          "id": 7,
          "name": "instrument"
        },
        {
          "name_string": "spanish",
          "id": 8,
          "name": "spanish"
        },
        {
          "name_string": "string",
          "id": 9,
          "name": "string"
        },
        {
          "name_string": "strings",
          "id": 10,
          "name": "strings"
        },
        {
          "name_string": "wooded",
          "id": 11,
          "name": "wooded"
        }
      ],
      "large_icon": {
        "file_extension": "jpeg",
        "url": "http://127.0.0.1:8001/api/image/12/",
        "id": 12,
        "security_marking": "UNCLASSIFIED"
      },
      "is_deleted": false,
      "avg_rate": 3,
      "total_rate4": 0,
      "approved_date": "2018-04-30T14:22:51.589990Z",
      "is_private": false,
      "description": "A guitar that produces sound acoustically by transmitting the vibration of the strings to the air as opposed to relying on electronic amplification. The sound waves from the strings of an acoustic guitar resonate through the guitar's body, creating sound.",
      "_score": 1,
      "agency_short_name": "Minipax",
      "approval_status": "APPROVED",
      "total_rate1": 1,
      "total_review_responses": 1,
      "title": "Acoustic Guitar",
      "is_enabled": true,
      "agency_id": 2,
      "banner_icon": {
        "file_extension": "jpeg",
        "url": "http://127.0.0.1:8001/api/image/13/",
        "id": 13,
        "security_marking": "UNCLASSIFIED"
      },
      "total_rate5": 1,
      "description_short": "A guitar that produces sound acoustically by transmitting the vibration of the strings.",
      "listing_type_id": 3,
      "listing_type_title": "Web Application",
      "small_icon": {
        "file_extension": "jpeg",
        "url": "http://127.0.0.1:8001/api/image/11/",
        "id": 11,
        "security_marking": "UNCLASSIFIED"
      },
      "total_rate3": 2,
      "security_marking": "UNCLASSIFIED",
      "listing_type_description": "web applications",
      "launch_url": "https://en.wikipedia.org/wiki/Acoustic_guitar",
      "id": 1,
      "categories": [
        {
          "title": "Music and Audio",
          "description": "Using your ears",
          "id": 10
        }
      ],
      "total_rate2": 0,
      "large_banner_icon": {
        "file_extension": "jpeg",
        "url": "http://127.0.0.1:8001/api/image/14/",
        "id": 14,
        "security_marking": "UNCLASSIFIED"
      },
      "agency_title": "Ministry of Peace",
      "is_508_compliant": true,
      "total_reviews": 4,
      "total_votes": 4
    }
  ],
  "previous": null,
  "next_offset_prediction": 24,
  "next": null,
  "count": 1
}
```

 
**How to test code (668)**    
```
make dev reindex_es use_es run
```

Validate setting up only_508_search_flag to false
```
curl -X PUT   http://127.0.0.1:8001/api/profile/1/   -H 'content-type: application/json'   -d '{"only_508_search_flag": false}' --user bigbrother:password
```

Check for Search Results
```
curl 'http://127.0.0.1:8001/api/listings/essearch/?search=&offset=0&limit=24'  --user bigbrother:password
```
There should be many results

Validate setting up only_508_search_flag to True
```
curl -X PUT   http://127.0.0.1:8001/api/profile/1/   -H 'content-type: application/json'   -d '{"only_508_search_flag": true}' --user bigbrother:password
```

Check for Search Results
```
curl 'http://127.0.0.1:8001/api/listings/essearch/?search=&offset=0&limit=24'  --user bigbrother:password
```
There should be 1 results




**Please Review**     
@aml-development/ozp-backend-team    

**Checklist**
